EIPS/eip-7685.md

```diff
@@ -13,83 +13,117 @@ created: 2024-04-14
 ## Abstract
 
 This proposal defines a general purpose framework for storing contract-triggered
-requests. It extends the execution header and body with a single field each to
-store the request information. This inherently exposes the requests to the
-consensus layer, which can then process each one.
+requests. It extends the execution header with a single field as a commitment to
+the request. Requests are later on exposed to the consensus layer.
+
+Requests are not stored on the execution block, only a commitment to the requests
+is stored in the execution block.
 
 ## Motivation
 
 The proliferation of smart contract controlled validators has caused there to be
 a demand for additional EL triggered behaviors. By allowing these systems to
 delegate administrative operations to their governing smart contracts, they can
 avoid intermediaries needing to step in and ensure certain operations occur.
-This creates a safer system for end users.
+This creates a safer system for end users. By abstracting each individual request
+details from the EL, adding new request types is simpler and does not require an
+update on the execution block structure.
 
 ## Specification
 
 ### Execution Layer
 
-#### Request
+#### Requests
+
+Let `request_data` be the output of a system call to a Request Smart Contract.
 
-A `request` consists of a `request_type` prepended to an opaque byte array
-`request_data`:
+`request_data` consists of two parts: `request_type` (single byte) and
+`request_list_ssz` (ssz list encoding of many requests) as following:
 
 ```
-request = request_type ++ request_data
+request_data = request_type ++ request_list_ssz
```

Comment:
It might worth mentioning a special case for an empty list, where `request_data = requests_type`, to make the spec more explicit